types/redux-first-router/index.d.ts

```diff
@@ -225,17 +225,17 @@ export function canGoBack(): boolean;
 
 export function canGoForward(): boolean;
 
-export function connectRoutes(
+export function connectRoutes<TKeys = {}, TState = any>(
     history: History,
-    routesMap: RoutesMap,
-    options?: Options
+    routesMap: RoutesMap<TKeys, TState>,
+    options?: Options<TKeys, TState>
 ): {
-    reducer: Reducer<LocationState>;
-    middleware: Middleware;
-    thunk(store: Store<any>): Promise<Nullable<RouteThunk>>;
-    enhancer: GenericStoreEnhancer;
-    initialDispatch?(): void;
-};
+        reducer: Reducer<LocationState<TKeys, TState>>;
+        middleware: Middleware;
+        thunk<ThunkState = TState>(store: Store<ThunkState>): Promise<Nullable<RouteThunk<ThunkState>>>;
```

Comment:
Is their a scenario where you would need to overwrite the State when using Thunk?

Reply:
No I can't really think of any! Could be replaced with a type parameter for the Action instead

Reply:
I would then just change it to:
`thunk(store: Store<TState>): Promise<Nullable<RouteThunk<TState>>>;`